packages/block-library/src/button/edit.js

```diff
@@ -209,7 +220,7 @@ function ButtonEdit( props ) {
 						setAttributes( { textAlign: nextAlign } );
 					} }
 				/>
-				{ ! isURLSet && (
+				{ ! isURLSet && 'a' === TagName && (
```

Comment:
Suggestion: The `'a' === TagName` check could be extracted into `isLinkTag` and reused in code.